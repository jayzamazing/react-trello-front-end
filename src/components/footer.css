.invisible {
  visibility: hidden;
}
footer {
    margin-top: 25px;
    width: 100%;
    height: 50px;
    text-align: center;
}
footer p {
    text-align: center;
}
.social {
  padding-left: 0;
  margin-left: auto;
  margin-right: auto;
  margin-bottom: 0;
}
.social-bottom, footer p {
  margin-bottom: 0;
  text-shadow: 2px 2px 3px black;
}
.social-bottom {
  max-width: 217px;
}
.social li {
    display: inline-block;
    margin: 5px;
}
footer, footer a, .genre label, .genre-submit input {
  color: white;
}
.social i, footer p, footer a {
  color: white;
}
