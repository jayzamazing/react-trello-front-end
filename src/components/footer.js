import React from 'react';
import './footer.css';
import {Link} from 'react-router-dom';

export function FooterBar(props){
  //only show when at the root page
  const root = props.location.pathname === '/' ? true : false;
  if (root) {
    return(
      <footer className="container footer-bar">
            <div className="col-xs-12 col-sm-12 col-md-12 col-lg-12">
              <ul className="social social-bottom">
                  <li><Link to="https://twitter.com/jayzamazing" target="_Blank"><i className="fa fa-twitter fa-3x" aria-hidden="true"></i></Link></li>
                  <li><Link to="https://github.com/jayzamazing" target="_Blank"><i className="fa fa-github fa-3x" aria-hidden="true"></i></Link></li>
                  <li><Link to="https://www.linkedin.com/pub/adrian-lopez/99/341/688" target="_Blank"><i className="fa fa-linkedin fa-3x" aria-hidden="true"></i></Link></li>
                  <li><Link to="https://plus.google.com/108258618753081021419" target="_Blank"><i className="fa fa-google-plus fa-3x" aria-hidden="true"></i></Link></li>
              </ul>
              <p>
                <span className="copyright">&#169; Adrian J Lopez</span>
              </p>
            </div>
      </footer>
    );
  } else {
    return null;
  }
}
