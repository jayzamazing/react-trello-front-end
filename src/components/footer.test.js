'use strict';
import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';
import {FooterBar} from './footer';

describe('footer', () => {
  it('should render page when home', () => {
    let location = {
      pathname: '/'
    }
    const wrapper = shallow(<FooterBar location={location}/>);
  });
  it('home snapshot', () => {
    let location = {
      pathname: '/'
    }
    const wrapper = shallow(<FooterBar location={location}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should not be visible', () => {
    let location = {
      pathname: '/other'
    }
    const wrapper = shallow(<FooterBar location={location}/>);
    expect(wrapper.find('footer')).toHaveLength(0);
  });
  it('should not be visible snapshot', () => {
    let location = {
      pathname: '/other'
    }
    const wrapper = shallow(<FooterBar location={location}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
