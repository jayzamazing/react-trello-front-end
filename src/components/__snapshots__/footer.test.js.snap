// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`footer home snapshot 1`] = `
<footer
  className="container footer-bar"
>
  <div
    className="col-xs-12 col-sm-12 col-md-12 col-lg-12"
  >
    <ul
      className="social social-bottom"
    >
      <li>
        <Link
          replace={false}
          target="_Blank"
          to="https://twitter.com/jayzamazing"
        >
          <i
            aria-hidden="true"
            className="fa fa-twitter fa-3x"
          />
        </Link>
      </li>
      <li>
        <Link
          replace={false}
          target="_Blank"
          to="https://github.com/jayzamazing"
        >
          <i
            aria-hidden="true"
            className="fa fa-github fa-3x"
          />
        </Link>
      </li>
      <li>
        <Link
          replace={false}
          target="_Blank"
          to="https://www.linkedin.com/pub/adrian-lopez/99/341/688"
        >
          <i
            aria-hidden="true"
            className="fa fa-linkedin fa-3x"
          />
        </Link>
      </li>
      <li>
        <Link
          replace={false}
          target="_Blank"
          to="https://plus.google.com/108258618753081021419"
        >
          <i
            aria-hidden="true"
            className="fa fa-google-plus fa-3x"
          />
        </Link>
      </li>
    </ul>
    <p>
      <span
        className="copyright"
      >
        Â© Adrian J Lopez
      </span>
    </p>
  </div>
</footer>
`;

exports[`footer should not be visible snapshot 1`] = `""`;
